tcms/testplans/tests/test_new_plan.py

```diff
@@ -96,16 +122,96 @@ def _test_plan_create_new(self, is_active):
+        new_data = self.request.copy()
+        new_data["product"] = -1
+        del new_data["email_settings-0-auto_to_plan_author"]
+
+        response = self.client.post(self.url, data=new_data, follow=True)
+
+        self.assertContains(response, _("Create new TestPlan"))
+        self.assertContains(
+            response,
+            _(
+                "Select a valid choice. That choice is not one of the available choices."
+            ),
+        )
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_plan_author" '
+            'type="checkbox"',
+            html=False,
+        )
+
+
+class TestNewTestPlanViewPermissions(PermissionsTestCase):
+    add_testplan_permission = "testplans.add_testplan"
+    permission_label = add_testplan_permission
+    http_method_names = ["get", "post"]
+    url = reverse("plans-new")
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.post_data = {"description": "test"}
+        super().setUpTestData()
+
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.plan_type = PlanTypeFactory()
+        cls.plan_name = TestPlanFactory()
+
+        cls.post_data.update(
+            {
+                "author": cls.tester.pk,
+                "product": cls.product.pk,
+                "product_version": cls.product_version.pk,
+                "type": cls.plan_type.pk,
+                "name": cls.plan_name.name,
+                "email_settings-0-auto_to_plan_author": "on",
+                "email_settings-0-auto_to_case_owner": "on",
+                "email_settings-0-auto_to_case_default_tester": "on",
+                "email_settings-0-notify_on_case_update": "on",
+                "email_settings-0-notify_on_plan_update": "on",
+                "email_settings-0-id": cls.plan_name.emailing.pk,
+                "email_settings-TOTAL_FORMS": "1",
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+                "is_active": True,
+            }
+        )
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('Create new TestPlan'))
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data, follow=True)
+        test_plan = TestPlan.objects.get(
+            name=self.post_data["name"],
+            product=self.post_data["product"],
+        )
+        redirect_url = reverse('test_plan_url', args=[test_plan.pk, slugify(test_plan.name)])
+
+        self.assertRedirects(response, redirect_url)
+        self._assert_test_plan(test_plan)
 
-        response = self.client.get(self.location, follow=True)
+    def verify_get_without_permission(self):
```

Comment:
this is already handled by PermissionTestCase itself and you rarely have to override it. Comment it out and if the tests pass then it is not needed. The same applies to verify_post_viothout_permission.